tasks/web_user_data

```diff
@@ -0,0 +1,179 @@
+
+apt-get update
+apt-get install -y awscli git nginx autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 libgdbm-dev
+
+sudo -u ubuntu git clone https://github.com/sstephenson/rbenv.git /home/ubuntu/.rbenv
+sudo -u ubuntu git clone https://github.com/sstephenson/ruby-build.git /home/ubuntu/.rbenv/plugins/ruby-build
+echo 'export PATH=$HOME/.rbenv/bin:$PATH' | sudo -u ubuntu tee -a /home/ubuntu/.bashrc
+echo 'eval "$(rbenv init -)"' | sudo -u ubuntu tee -a /home/ubuntu/.bashrc
+
+echo 'export NODE_ENV=production' | sudo -u ubuntu tee -a /home/ubuntu/.bashrc
+sudo -u ubuntu curl https://raw.githubusercontent.com/creationix/nvm/v0.31.2/install.sh | sudo -u ubuntu bash
+
+sudo -u ubuntu mkdir -p /home/ubuntu/oversight/current
+sudo -u ubuntu mkdir -p /home/ubuntu/oversight/shared/log
+sudo -u ubuntu mkdir -p /home/ubuntu/oversight/shared/sitemap
+cd /home/ubuntu/oversight/current
+sudo -u ubuntu git clone https://github.com/konklone/oversight.garden.git .
+
+sudo -u ubuntu bash <<'END'
+export PATH=$HOME/.rbenv/shims:$HOME/.rbenv/bin:$PATH
+rbenv install 2.2.5
+rbenv global 2.2.5
+gem install bundler
+rbenv rehash
+bundle install
+rake letsencrypt:fetch
+END
+
+sudo -u ubuntu bash <<'END'
+export NVM_DIR=$HOME/.nvm
+source $NVM_DIR/nvm.sh
+nvm install 4.4.7
+nvm alias default 4.4.7
+npm install --production
+./node_modules/.bin/wintersmith build --config=config/blog.js
+END
+
+bash <<'END'
+export NVM_DIR=/home/ubuntu/.nvm
+source $NVM_DIR/nvm.sh
+npm install -g forever
+END
+
+sudo -u ubuntu cp config/config.yaml.example ../shared/config.yaml
+sed -i 's/127.0.0.1/search-oversight-iu2rwt666hmlxrbn2slw6zlgou.us-east-1.es.amazonaws.com/' ../shared/config.yaml
+sed -i 's/9200/80/' ../shared/config.yaml
+sed -i 's_  # directory: /path/to/sitemaps_  directory: /home/unitedstates/oversight/shared/sitemap_' ../shared/config.yaml
+echo 'aws:' >> ../shared/config.yaml
+echo '  region: us-east-1' >> ../shared/config.yaml
+sudo -u ubuntu aws s3 cp s3://oversight-secrets/secrets.sh /home/ubuntu/
+sudo -u ubuntu bash /home/ubuntu/secrets.sh ../shared/config.yaml
+sudo -u ubuntu ln -s /home/ubuntu/oversight/shared/config.yaml config/config.yaml
+
+sudo -u ubuntu bash <<'END'
+export NVM_DIR=/home/ubuntu/.nvm
+source $NVM_DIR/nvm.sh
+export NODE_ENV=production
+forever -l /home/ubuntu/oversight/forever.log -a start app.js
+END
+
+# Create nginx config
+cat >/etc/nginx/ssl.rules <<'END'
+ssl_certificate /home/ubuntu/letsencrypt.fullchain.pem;
+ssl_certificate_key /home/ubuntu/letsencrypt.privkey.pem;
+
+# always use http/2 over 443
+listen 443 ssl http2;
+
+# This requires strong forward secrecy (ECDHE) for all connections.
+# However, it blocks IE8+XP and Android 2.3.
+ssl_ciphers 'kEECDH+ECDSA+AES128 kEECDH+ECDSA+AES256 kEECDH+AES128 kEECDH+AES256 +SHA !aNULL !eNULL !LOW !MD5 !EXP !DSS !PSK !SRP !kECDH !CAMELLIA !RC4 !SEED';
+
+# cut out SSLv3
+ssl_protocols TLSv1.2 TLSv1.1 TLSv1;
```

Comment:
A note that, depending on traffic, I'd be up for potentially cutting out TLSv1.1 and TLSv1.0 at some point.